### PR TITLE
[LF] Remove last remaining of Decimal

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
Decimal stays a alias for Numeric 10 in Daml.

### CODE REVIEW

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinBigNumericTest.scala [hunk 1]
```diff
@@ -45,8 +45,8 @@ class SBuiltinBigNumericTest(majorLanguageVersion: LanguageMajorVersion)
     else
       "0." + "0" * (-i - 1) + "1" + "0" * (scale + i)
 
-  private val decimals = Table[String](
-    "Decimals",
+  private val numerics = Table[String](
+    "Nueric",
```
Comment (paulbrauner-da) [inline daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinBigNumericTest.scala:49]: typo: missing 'm'

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinTest.scala [hunk 2]
```diff
@@ -431,7 +431,7 @@ class SBuiltinTest(majorLanguageVersion: LanguageMajorVersion)
       "returns proper result" in {
         val d = "8765432109876543210987654321.0987654321"
         val testCases = Table[Long, String, String](
-          ("rounding", "decimal", "result"),
+          ("rounding", "nuemric", "result"),
```
Comment (paulbrauner-da) [inline daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinTest.scala:434]: typo: em->me

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/crypto/HashSpec.scala [hunk 3]
```diff
@@ -352,9 +353,9 @@ class HashSpec extends AnyWordSpec with Matchers {
       val units = List(ValueUnit)
       val bools = List(true, false).map(VA.bool.inj(_))
       val ints = List(-1L, 0L, 1L).map(VA.int64.inj(_))
-      val decimals = List("-10000.0000000000", "0.0000000000", "10000.0000000000")
+      val numric10s = List("-10000.0000000000", "0.0000000000", "10000.0000000000")
```
Comment (paulbrauner-da) [inline daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/crypto/HashSpec.scala:356]: typo: missing 'e'

**Review by paulbrauner-da — APPROVED** [review]
Nice!

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinBigNumericTest.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinBigNumericTest.scala:49]:
```suggestion
    "Numeric",
```

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinTest.scala (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinTest.scala:434]:
```suggestion
          ("rounding", "numeric", "result"),
```

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/crypto/HashSpec.scala (hunk 3 above)
Comment (remyhaemmerle-da) [reply daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/crypto/HashSpec.scala:356]:
```suggestion
      val numeric10s = List("-10000.0000000000", "0.0000000000", "10000.0000000000")
```

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/crypto/HashSpec.scala [hunk 4]
```diff
@@ -473,7 +474,7 @@ class HashSpec extends AnyWordSpec with Matchers {
           List(Some(None), Some(Some(false))).map(VA.optional(VA.optional(VA.bool)).inj(_))
 
       val testCases: List[V] =
-        units ++ bools ++ ints ++ decimals ++ numeric0s ++ dates ++ timestamps ++ texts ++ parties ++ contractIds ++ optionals ++ lists ++ textMaps ++ genMaps ++ enums ++ records0 ++ records2 ++ variants
+        units ++ bools ++ ints ++ numric10s ++ numeric0s ++ dates ++ timestamps ++ texts ++ parties ++ contractIds ++ optionals ++ lists ++ textMaps ++ genMaps ++ enums ++ records0 ++ records2 ++ variants
```
Comment (remyhaemmerle-da) [inline daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/crypto/HashSpec.scala:477]:
```suggestion
        units ++ bools ++ ints ++ numeric10s ++ numeric0s ++ dates ++ timestamps ++ texts ++ parties ++ contractIds ++ optionals ++ lists ++ textMaps ++ genMaps ++ enums ++ records0 ++ records2 ++ variants
```

**Pull request merged**